### ISSUE TITLE
Spot-to-forward conversion for the payoff x-axis

### ISSUE THREAD

**Issue by lalitkarthik** [lalitkarthik/convex-hedge-payoff#13]
Part of #1

## Question

Graduated from fog by #2, which established that the engine prices with **Black-76 on the forward**, not on spot.

That creates a problem the map did not previously have. The payoff chart's x-axis is **spot** — the user drags a slider to "what if NIFTY is at 23,070". But the pricer needs a **forward**. So every x-value has to be converted, and there is no obvious right answer.

Settle:

1. **How do we get a forward for a hypothetical spot?** Options: hold the carry constant and set `F = S / D` (using today's `discount` at the target date's `T`); hold the *basis* constant and set `F = S + (F₀ − S₀)`; or hold the *ratio* constant, `F = S · (F₀ / S₀)`. These disagree by a few points, which is small in absolute terms but shifts the whole payoff curve.
2. **Which rate do we use, and where does it come from?** #2 found the implied carry `r = −ln(D)/T` is noisy and unstable, running from 0.9% to 28.4% across the dataset and blowing up as `T → 0`. Do we fit a stable rate per expiry, take an ATM-strike reading, hardcode a sensible figure, or something else?
3. **What is the x-axis actually showing?** The Sensibull screenshot labels it with spot values (23,500 / 24,000 / 24,500) while displaying spot 24,043.20 and futures 24,103.10 separately. Sixty points apart. Confirm the axis is spot and that the conversion happens internally.
4. **Does this matter at expiry?** At `T = 0` the forward converges to the spot and the discount is 1, so the green line is unaffected. It only bites on the blue target-date line — which means an implementation could be quietly wrong in a way that only shows up in one of the two curves.
5. **Does the same conversion apply to breakevens and POP?** Breakevens are quoted in spot terms in the screenshot (23,600 and 24,200), but POP is computed from a lognormal in forward terms. Getting these in different units would produce a plausible-looking but wrong POP.

**Blocked by** the data-slice ticket — deciding this needs the spot series from `index.parquet` joined against `forward` from `greeks.parquet`, so the basis can be looked at directly rather than assumed.

## Answer

_(record on resolution)_

**Comment by lalitkarthik** [comment]
## Partial finding from the research notebook (PR #15)

One candidate rule is now **ruled out**, empirically rather than by argument.

The notebook first tried holding the ratio constant, `F(S) = S · (F₀/S₀)`. An assertion that `payoff_at_date(T=0)` must equal `payoff_at_expiry` failed by **123 points**.

The reason generalises: **any valid rule must satisfy `F → S` as `T → 0`.** On expiry day there is no carry left to pay, so the forward and the spot are the same number. Freezing the ratio leaves the whole 119-point basis in place at expiry, which shifts the entire green line sideways.

That kills the constant-ratio rule and constrains the remaining two:

| Rule | `F → S` as `T → 0`? |
|---|---|
| Constant ratio, `F = S·(F₀/S₀)` | **No** — ruled out |
| Constant carry rate, `F = S·exp(r_c·T)` where `r_c = ln(F₀/S₀)/T₀` | Yes |
| Constant additive basis, `F = S + (F₀ − S₀)` | **No** — same failure |
| Decaying additive basis, `F = S + (F₀ − S₀)·(T/T₀)` | Yes |

So the real choice is narrower than the ticket assumed: **constant carry rate versus decaying additive basis**. Both converge correctly; they differ in how the basis behaves for spots far from today's.

The notebook uses constant carry rate. Measured on this snapshot: `r_c = 11.27%`, against an implied funding rate of `15.61%` from the discount factor. That gap is itself interesting and probably the dividend yield — worth checking against the `NIFTY50 DIV POINT` series when this ticket is worked properly.

**Recommendation for whoever takes this**: keep the `F → S` assertion as a permanent test regardless of which rule wins. It is one line and it catches the entire class of error.

**Comment by lalitkarthik** [comment]
## Evidence for Q1 and Q2, measured on the full dataset

Joined `greeks.parquet` (UTC) to the `NIFTY 50.NSE_IDX` series in `index.parquet` using the #6 rule (floor to the minute, shift IST by 5h30m). **All 8,356 greeks timestamps matched a spot bar, 0 unmatched.**

### The basis is not a constant, and it is not always positive

| | Basis (F − S) |
|---|---|
| mean | +46.18 |
| median | +20.39 |
| min | **−26.45** |
| max | +233.04 |
| minutes with negative basis | **359** |
| minutes with `forward == spot` exactly | **2,397 of 8,356 (28.7%)** |

So "hold the basis constant" (`F = S + (F₀ − S₀)`) is being asked to hold constant a quantity that ranges over 260 points and changes sign.

### It converges, which is the constraint that matters

Median basis by days to expiry:

| dte | median basis |
|---|---|
| 7–14 | +98.42 |
| 3–7 | +21.16 |
| 1–3 | +8.35 |
| 0–1 | **−2.40** |

This is the empirical version of the assertion #9 already carries — **F → S as T → 0**. Any conversion rule we pick has to reproduce this, and the constant-basis rule does not: it would still be adding ~98 points on expiry morning.

### The 28.7% fallback is a chain-thickness problem, not a time problem

The 2,397 minutes where `forward == spot` exactly quote a **median of 7 strikes**, against **87 strikes** on the healthy minutes. The parity regression needs a spread of strikes to fit; with 7 it gives up and returns spot. That is a data-coverage artefact of the early weeks, not something the engine will hit on a live chain — but it does mean **Q2's "where does the rate come from" cannot be answered by reading `discount` blindly**, because on those minutes it is not a fitted number.

Recommendation for Q1 shifts accordingly: **constant ratio, `F = S · (F₀/S₀)`** — it satisfies F → S at T → 0 by construction, survives a sign change in the basis, and is what `notebooks/01_payoff_structures.ipynb` already does (`CARRY_RATE = ln(F/S)/T`). Constant-basis is ruled out by the table above.

**Comment by lalitkarthik** [comment]
## Correction to my earlier comment

My earlier comment recommended **constant ratio**, `F = S·(F₀/S₀)`, and claimed it "satisfies F → S at T → 0 by construction" and is "what the notebook already does". **Both claims are wrong.**

Constant ratio does the opposite: it never converges. `F₀/S₀ = 1.004736` at this snapshot, and it stays 1.004736 forever, so at expiry it puts the forward **118.87 points above** the spot when the correct answer is zero.

And the notebook does **not** do that. It computes `CARRY_RATE = ln(FORWARD / SPOT) / T` once, then `forward = spot * exp(CARRY_RATE * T_left)` — that is **constant carry rate**, a different rule, and it is the one that converges.

### The test that separates them

Both rules are calibrated to the same observation: spot 25,100.25, forward 25,219.12, 10.56 trading days left, implying a carry of 11.2743% per year. Ask each what the forward is at expiry, when there is no carry left to pay:

| rule | F at T_left = 0 | gap to spot |
|---|---|---|
| constant ratio, `F = S·(F₀/S₀)` | 25,219.12 | **+118.87** ❌ |
| constant carry, `F = S·e^{r_c·T}` | 25,100.25 | **+0.00** ✅ |

They also disagree in between — at 5 trading days left, by **62.65 points**.

### What it costs downstream

Priced through a long straddle at 25200, the constant-ratio rule corrupts the **expiry** line, which should not depend on any conversion at all:

| spot | correct P&L at expiry | with the ratio rule | error |
|---|---|---|---|
| 25,100 | −570.75 | −651.89 | −81.14 |
| 25,200 | −670.75 | −551.41 | **+119.34** |

The straddle is supposed to lose exactly its premium of 670.75 at the strike. Under the ratio rule it reads −551.41, because the pricer is handed a forward that is still 119 points above the spot on expiry day.

### Standing recommendation

**Constant carry rate**, `F(S,T) = S·e^{r_c·T}` with `r_c = ln(F₀/S₀)/T₀`. It satisfies `F → S as T → 0` by construction, which is the assertion #9 already carries, and it is what the prototype implements.

**Q1 is therefore effectively settled and constant-basis is out.** What remains genuinely open is **Q2**: `r_c` here is calibrated per snapshot from that minute's own forward, which is fine for a single chart but says nothing about a stable per-expiry rate. Also still open is the additive-basis alternative, which decays the basis linearly rather than exponentially and gives a slightly different curve while still converging.

Apologies for the noise — the earlier comment's evidence tables were right and its conclusion did not follow from them.

**Comment by MeghavPradeep** [comment]
Evidence toward this, from #44 — **not an answer**, and the issue stays open.

#44 rewrites the payoff notebook's maths in the forward and draws every structure at both anchors. Two things it settles empirically:

**Question 4 — "does this matter at expiry?"** Confirmed, concretely. At expiry `F_T = S_T`, so the expiry payoff can be written in `F` with no conversion at all. That is now the notebook's notation: `F` is the payoff variable and the x-axis, `F̂` is the forward fitted in §3. Since the expiry curve is the only line v1 draws, the conversion question genuinely does not bite yet — exactly as this issue predicted. It bites on the target-date line, which is still blocked behind #8.

**Question 3 — "what is the x-axis actually showing?"** Worth separating two things that are easy to conflate:

- the **x-axis** conversion — what this issue is about, and still open
- the **strike selection** anchor — which strike is "at the money"

The second is not blocked and does matter now. At the 12:00 snapshot the basis is **+118.87 points**, and the two anchors pick different strikes: **25,100** from spot, **25,200** from the forward. #44 builds both and puts them side by side. They are different positions, not two renderings of one — every premium, cap and breakeven moves:

| | net (fwd) | net (spot) | Δ |
|---|---|---|---|
| Short Straddle | −43,599 | −44,353 | +754 |
| Bull Call Spread | 8,960 | 12,750 | −3,790 |
| Iron Fly | −16,672 | −19,243 | +2,571 |
| Iron Condor | −10,949 | −8,573 | −2,376 |

`docs/calculations.md` already states the forward is the correct anchor. This just shows the size of being wrong about it.

**A related trap, found while doing it.** The anchor choice can silently change a structure's *shape*, not only its prices. **25,400 CE is not quoted** at this minute, so at the spot anchor the iron fly's call wing lands at 25,500 — 400 points out against 300 on the put side. `max loss = credit − wL` is then wrong; it needs `credit − max(w_left, w_right)·L`. Under the forward anchor everything comes out symmetric, so this never showed up before. Whatever conversion rule question 1 lands on, outer legs have to be placed at **quoted** strikes and the width read back off the strikes actually used, never assumed from the nominal offset.

Still open here and untouched by #44: **question 1** (which of `F = S/D`, constant basis, or constant ratio), **question 2** (which rate), and **question 5** (breakevens and POP sharing units).

**Comment by MeghavPradeep** [comment]
Further evidence from #45, which reframes the payoff notebook onto two coordinate systems. Still **not an answer**; the issue stays open.

**Question 1 — one of the three rules is excluded on this data.**

```
F = S / D          ->  25,264.98     observed 25,219.12     miss +45.86
F = S + b          ->  25,219.12     exact by construction
F = S x (F̂/S)     ->  25,219.12     exact by construction
```

`S/D` reinstates the full financing cost and none of the offsetting dividend income, because the fitted `D` is a **pure discount**. The 45.86-point miss is a strike interval wide — larger than the basis it is trying to reproduce. So the choice is between constant basis and constant ratio, which disagree by under 7 points across ±6% of spot:

```
spot        basis rule      ratio rule       gap
23,600      23,718.87       23,711.76      +7.10
25,100      25,218.87       25,218.86      +0.00
26,600      26,718.87       26,725.97      -7.10
```

#45 adopts **constant basis for charting**, on a provability argument rather than an accuracy one: it makes the transform a rigid translation, `Π_F(F) = Π(F − b)`, so the caps can be asserted *unchanged* and the breakevens asserted to move by *exactly* `b`. The ratio rule scales, stretching a 300-point spread to 301.42, and every width and cap would need re-deriving. That is a chart-axis convention, not a pricing decision, and questions 2 and 5 are untouched.

**Question 4 — confirmed, and it cuts both ways.** At expiry `F_T = S_T`, so the expiry line needs no conversion. But that is exactly what makes a forward-axis chart *empty* unless the strikes are re-expressed: `max(S_T − K, 0) = max(F − (K + b), 0)`, so in forward coordinates the effective strike is `K + b`. Without that step the "forward" chart is the spot chart with a relabelled axis.

**A trap worth recording before anyone implements the conversion in the engine.** The candidate set for the piecewise-linear extrema is `{0} ∪ {Kᵢ}` in spot coordinates. In forward coordinates it becomes `{b} ∪ {Kᵢ + b}` — the left tail terminates at `F = b`, because `S_T ≥ 0`. Reusing `0` evaluates the payoff at a spot of −118.87. On the naked long put:

```
correct   max profit at F = b   ->  1,616,764.50
naive     max profit at F = 0   ->  1,624,490.77     wrong by +7,726.27
```

No exception, no `NaN`, 0.48% high. Whatever rule question 1 lands on, the safe shape is to compute metrics **once in spot coordinates**, where the floor is unambiguously zero, and translate only the x-quantities — so no second candidate set exists to get wrong.

Still open and untouched by #45: **question 2** (which rate), **question 5** (breakevens and POP sharing units), and question 1 itself, now down to two candidates.
